Hashids/Hashids.swift

```diff
@@ -50,230 +48,205 @@ public typealias Hashids = Hashids_<UInt32>
-            hash = alphabet[rrange] + hash + alphabet[lrange];
+        let half_length = alphabet.count >> 1
+        while( hash.count < minLength ) {
+            shuffle(&alphabet, salt: alphabet)
+            let lrange = 0..<half_length
+            let rrange = half_length..<(alphabet.count)
+            hash = alphabet[rrange] + hash + alphabet[lrange]
             
-            let excess = hash.count - minLength;
+            let excess = hash.count - minLength
             if( excess > 0 )
             {
-                let start = excess >> 1;
+                let start = excess >> 1
                 hash = [Char](hash[start..<(start+minLength)])
             }
         }
         
-        return hash;
+        return hash
     }
     
-    private func _decode(hash:[Char]) -> [Int]
-    {
-        var ret = [Int]();
+    private func _decode(hash: [Char]) -> [Int] {
+        var ret = [Int]()
         
-        var alphabet = self.alphabet;
+        var alphabet = self.alphabet
         
         var hashes = hash.split(hash.count, allowEmptySlices: true) {
             self.guards.contains($0)
         }
         
-        let hashesCount = hashes.count, i = ((hashesCount == 2) || (hashesCount == 3)) ? 1 : 0;
-        let hash = hashes[i];
+        let hashesCount = hashes.count, i = ( hashesCount == 2 || hashesCount == 3) ? 1 : 0
+        let hash = hashes[i]
         
-        if !hash.isEmpty
-        {
-            let lottery = hash[hash.startIndex];
-            let valuesHashes = hash[1..<hash.count];
-           let valueHashes = valuesHashes.split(valuesHashes.count, allowEmptySlices: true) { self.seps.contains($0) }
-            var lsalt = [Char]();
-            let (lsaltARange, lsaltRange) = _saltify(&lsalt, lottery, alphabet);
+        if !hash.isEmpty {
+            let lottery = hash[hash.startIndex]
+            let valuesHashes = hash[1..<hash.count]
+            let valueHashes = valuesHashes.split(valuesHashes.count, allowEmptySlices: true) { self.seps.contains($0) }
+            var lsalt = [Char]()
+            let (lsaltARange, lsaltRange) = _saltify(&lsalt, lottery, alphabet)
 
-            for subHash in valueHashes
-            {
-                shuffle(&alphabet, salt: lsalt, saltRange: lsaltRange);
-                ret.append(self._unhash(subHash, alphabet));
-                lsalt.replaceRange(lsaltARange, with: alphabet);
+            for subHash in valueHashes {
+                shuffle(&alphabet, salt: lsalt, saltRange: lsaltRange)
+                ret.append(self._unhash(subHash, alphabet))
+                lsalt.replaceRange(lsaltARange, with: alphabet)
             }
         }
         
-        return ret;
+        return ret
     }
     
-    private func _hash(inout hash:[Char], var _ number:Int, _ alphabet:[Char])
-    {
-        let length = alphabet.count, index = hash.count;
+    private func _hash(inout hash: [Char], var _ number: Int, _ alphabet: [Char]) {
```

Comment:
While your'e doing all this, it might be worth re-evaluating if this should be dropping the parameter names.